.travis.yml

```diff
@@ -1,48 +1,65 @@
-# (C) Copyright 2009-2019 Enthought, Inc., Austin, TX
+# (C) Copyright 2009-2020 Enthought, Inc., Austin, TX
 # All rights reserved.
 
 language: generic
-
+dist: xenial
 services:
   - xvfb
 
 addons:
   apt:
     packages:
+      - graphviz
       - libglu1-mesa-dev
+      - libxkbcommon-x11-0
+      - libxcb-icccm4
+      - libxcb-image0
+      - libxcb-keysyms1
+      - libxcb-randr0
+      - libxcb-render-util0
+      - libxcb-xinerama0
 
 env:
   global:
-    - INSTALL_EDM_VERSION=2.0.0
-    - ETS_TOOLKIT=qt4
+    - INSTALL_EDM_VERSION=3.0.1
+      PYTHONUNBUFFERED="1"
+      QT_DEBUG_PLUGINS="1"
 
 matrix:
   include:
-      - env: RUNTIME=2.7 QT_API=pyqt NX_VERSION=-1.11-7
-      - env: RUNTIME=2.7 QT_API=pyqt
-      - env: RUNTIME=2.7 QT_API=pyside NX_VERSION=-1.11-7
-      - env: RUNTIME=2.7 QT_API=pyside
-      - env: RUNTIME=3.5 QT_API=pyqt NX_VERSION=-1.11-7
-      - env: RUNTIME=3.5 QT_API=pyqt
-      - env: RUNTIME=3.6 QT_API=pyqt NX_VERSION=-1.11-7
-      - env: RUNTIME=3.6 QT_API=pyqt
+  - env: RUNTIME=3.6 TOOLKITS="pyqt pyqt5 pyside2"
+  fast_finish: true
 
 cache:
   pip: true
   directories:
+    - $HOME/.cache
     - $HOME/.ccache
 
 before_install:
   - mkdir -p "${HOME}/.cache/download"
-  - export DISPLAY=:99.0
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ./install-edm-linux.sh; export PATH="${HOME}/edm/bin:${PATH}"; export PLATFORM_TAG="rh6-x86_64"; fi
-  - echo ${PATH}
-  - edm environments create python${RUNTIME} --version ${RUNTIME} --platform ${PLATFORM_TAG}
+  - if [[ ${TRAVIS_OS_NAME} == "linux" ]] ; then
+       ccache -s;
+       ./ci/install-edm-linux.sh;
+       export PATH="${HOME}/edm/bin:/usr/lib/ccache:${PATH}";
+    else
+       ./ci/install-edm-osx.sh;
```

Comment:
We don't have macos in the matrix so this never gets executed. Do you want to add macos to the matrix? If not, we can just remove this branch.

Reply:
Definitely don't want to run MacOS on TravisCI given the costs at this point. I've removed this branch in 2b57766.